### PR TITLE
[Pylint] Check client methods have approved name prefix

### PR DESCRIPTION
**Pull request by MJoshuaB**
Refactors existing checker `ClientHasApprovedMethodNamePrefix` to better fit guidelines, remove false positives, and ignore internal modules and class properties.

Affected SDKs:
 - `azure.cognitiveservices.language.textanalytics.text_analytics_client` 3 errors
 - `azure.cognitiveservices.personalizer.personalizer_client` 1 error

Includes some extra common prefixes/verbs: close, cancel, clear, subscribe, send, query
Includes prefixes/verbs for future use: analyze, train, detect

### CODE REVIEW

File: tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py [hunk 1]
```diff
@@ -3597,8 +3532,62 @@ def test_extra_nested_branches_exception_logged(self, setup):
         ):
             self.checker.visit_try(try_node)
 
+
+class TestCheckDoNotUseLegacyTyping(pylint.testutils.CheckerTestCase):
```
Comment (l0lawrence) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py:3536]: can we remove this looks like it got added in a merge :)

**Review by l0lawrence — COMMENTED** [review]
looks good :)

File: tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py (hunk 1 above)
Comment (MJoshuaB) [reply tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py]: It has been removed.

File: tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py [hunk 2]
```diff
@@ -302,202 +303,131 @@ def test_guidelines_link_active(self):
         assert response.http_response.status_code == 200
 
 
+def _load_file(filename):
```
Comment (l0lawrence) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py:306]: oh nice :)

**Review by l0lawrence — APPROVED** [review]
lgtm thanks!

File: tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py (hunk 2 above)
Comment (MJoshuaB) [reply tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py:306]: Thank you, it allowed for some of my tests to be simplified.

**Pull request merged**